### ISSUE TITLE
service archiver does not warn for v10+

### ISSUE THREAD

**Issue by ioguix** [OPMDG/check_pgactivity#236]
For v10 and after, the archiver service only check the number of waiting WAL. It does not report when the archiver is currently failing. This is a bug. This bug is especially bad if there's no threshold set up...

**Comment by ioguix** [comment]
Moreover:

- when the archiver is failing, the `oldest_archive_failure` is actually the lastest one...
- when the archiver work correctly, the query doesn't report the number of WAL waiting for archiving.

This is completely broken.

**Comment by frost242** [comment]
Hello,

Since the fix, it seems that there are still some things broken with check_archiver.

When the archiver has never managed to archive successfully a WAL file, the checker returns OK : 
```
$ /usr/lib/nagios/plugins/check_pgactivity -s archiver -F human -h /var/run/postgresql -p xxxx
Argument "" isn't numeric in numeric gt (>) at /usr/lib/nagios/plugins/check_pgactivity line 1949.
Service        : POSTGRES_ARCHIVER
Returns        : 0 (OK)
Message        : 0 WAL files ready to archive
Perfdata       : ready_archive=0 min=0
Perfdata       : oldest_ready_wal=0s min=0
```

In this case, the stats sub-query returns the following : 
```
-[ RECORD 1 ]------+-------------------------
last_archived_wal  |
last_archived_time |
last_failed_wal    | 0000000100000000000000FC
walsegsize         | 16777216
last_archived_off  |
current_pos        | 0/FE000110
failing            | t
```

The query itself returns this :
```
-[ RECORD 1 ]-----+-------------------------
coalesce          | 0
case              |
last_archived_wal |
last_failed_wal   | 0000000100000000000000FC
oldest            |
```

For the record : 
```
# SELECT pg_current_wal_lsn();
-[ RECORD 1 ]------+-----------
pg_current_wal_lsn | 0/FE000110
```

A first problem lies in this part of the main query : 
```
CASE WHEN failing
  THEN extract('epoch' from (current_timestamp - last_archived_time))
  ELSE 0
END
```
As last_archived_time is NULL, this leads to the following runtime error :
`Argument "" isn't numeric in numeric gt (>) at /usr/lib/nagios/plugins/check_pgactivity line 1949.`
And of course, this check does not execute the according codepath for a failing archiver.

**Comment by frost242** [comment]
Opened new issue #290 for better tracking.
